go/service/main.go

```diff
@@ -255,7 +255,9 @@ func (d *Service) createMessageDeliverer() {
 	si := func() libkb.SecretUI { return chat.DelivererSecretUI{} }
 	ti := func() keybase1.TlfInterface { return tlf }
 
-	sender := chat.NewBlockingSender(d.G(), chat.NewBoxer(d.G(), ti), d.attachmentstore, ri, si)
+	bodyHashChecker := storage.ChatDbBackedBodyHashChecker(d.G())
```

Comment:
I had some in-memory shims for this before, but I just removed them, so yeah I think we can un-parametrize this.